fastn-core/src/commands/serve.rs

```diff
@@ -489,6 +493,26 @@ async fn handle_static_route(
     }
 }
 
+async fn handle_migrations(
+    config: &fastn_core::Config,
+    req: &fastn_core::http::Request,
+) -> Option<fastn_core::Result<fastn_core::http::Response>> {
+    if !req.path().eq("migration") {
+        return None;
+    }
+
+    let mut req_config = fastn_core::RequestConfig::new(config, &req, "", "/");
+
+    let db = fastn_core::migrations::get_db_url(config).await;
+    fastn_core::migrations::create_migration_table(config, db.as_str()).await?;
+
+    fastn_core::migrations::migrate(&mut req_config, db.as_str()).await?;
+
+    Some(Ok(actix_web::HttpResponse::build(actix_web::http::StatusCode::OK)
+        .append_header(("Content-Type", "application/json"))
+        .body(serde_json::json!({"status": "success"}).to_string())))
+}
```

Comment:
Nopes. Before we process any page we will run migration (this is the best we can do, ideally we should only run migration if migration section has changed, or we load the site for the first time etc, not sure how can we do it in fastn, maybe we can do this kind of optimisation in hostn).

Reply:
Earlier we were planning to run migration when the [user visits /migration/ url](https://github.com/orgs/fastn-stack/discussions/1881)

Reply:
Storing it in FASTN.ftd is better design as otherwise people will have to explicitly go to that URL to triggger migraiton.

Reply:
@amitu But this saves us from running migrations repeatedly and helps us do more expensive checks because this won't be happening so frequently.